### PR TITLE
Class Accuracy Metric Plugin.

### PR DESCRIPTION
**Pull request by VerwimpEli**
First version of a Class Accuracy Metric Plugin, such that keeping track of the accuracy per class is possible. If the overall idea is okay, I'll add tests and some convenience wrappers, similarly to the mean Accuracy Metric Plugin (and move the `__str__` method to those classes).

Some thoughts:
* Returning 0.0 as the default value is difficult in this case, since only the classes seen in the current eval loop are tracked. So after `reset`, `_class_accuracies` is just an empty dict. If this is unwanted behavior, we could initialize the dict with a 0.0 accuracy value after a rest, but I didn't see a proper way to get all class labels from within the Metric definition.
* Converting and checking for one hot vectors should maybe moved to a util method? Code duplicates the `Accuracy` metric partly now. 
*  WandB logger can't handle dictionaries at the moment. So that should be implemented perhaps after this PR is merged, if the dict persists.

### CODE REVIEW

**Comment by AndreaCossu** [comment]
Hi @VerwimpEli! It is fine if the metric returns an empty dictionary in case no class has been encountered, yet. W&BLogger will need to be updated anyway because in my last PR I also added metric returning dictionaries, so no worries.

Feel free to move the one-hot converter so that the code can be reused across metrics :smile:

**Review by Mathieu4141 — COMMENTED** [review]
It's a cool metric to have 😄 

> WandB logger can't handle dictionaries at the moment. So that should be implemented perhaps after this PR is merged, if the dict persists.

I think it is also true for the Tensorboard logger

File: avalanche/evaluation/metrics/class_accuracy.py [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+                continue
+
+            mask = torch.eq(true_y, label)
+            try:
+                self._class_accuracies[label].update(predicted_y[mask],
+                                                     true_y[mask])
+            except KeyError:
+                self._class_accuracies[label] = Accuracy()
+                self._class_accuracies[label].update(predicted_y[mask],
+                                                     true_y[mask])
```
[the remark this answers is not in the document]
Comment (VerwimpEli) [reply avalanche/evaluation/metrics/class_accuracy.py]: @Mathieu4141 cool, somehow I didn't know that. Thanks! Updated the PR :)

**Comment by vlomonaco** [comment]
@VerwimpEli @AndreaCossu and @Mathieu4141 is this ready to merge after we resolve the conflict in the test?

**Comment by VerwimpEli** [comment]
@vlomonaco it's ready!

**Comment by AndreaCossu** [comment]
I'll take care of the conflict and merging

**Comment by AndreaCossu** [comment]
Hi @VerwimpEli ! New changes in the metrics raised some errors with this PR. The good news is that it shouldn't take long to adapt your code. 
Please, take a look at the new `Accuracy` class, which now requires `task_label` in the `update` call. You can check out the plugin metrics for accuracy to understand how to get the task label information. 

For example from the `AccuracyPluginMetric` you can take this piece of code:
```python
def update(self, strategy):
    # task labels defined for each experience
    task_labels = strategy.experience.task_labels
    if len(task_labels) > 1:
        # task labels defined for each pattern
        task_labels = strategy.mb_task_id
    else:
        task_labels = task_labels[0]
    self._accuracy.update(strategy.mb_output, strategy.mb_y, task_labels)
```

Let me know if you encounter any problem :smile:

**Comment by AndreaCossu** [comment]
Hi @VerwimpEli I see you recently pushed a fix for task labels. Is this ready to be merged? I can take care of the conflicts.

**Comment by VerwimpEli** [comment]
Hi @AndreaCossu sadly no, it's just a place holder solution for now because we use my fork of Avalanche in our [ICCV challenge](https://sslad2021.github.io/pages/challenge.html). Will try to make a definitive solution soon!

**Comment by lrzpellegrini** [comment]
Hi @VerwimpEli, the metrics have been merged with PR #1051 (for which I took inspiration from your proposal). Thank you!

**Pull request closed without merging**